prov/terraform/openstack/osinstance.go

```diff
@@ -343,18 +347,17 @@ func (g *osGenerator) generateOSInstance(ctx context.Context, kv *api.KV, cfg co
 	} else {
 		accessIP = "${openstack_compute_instance_v2." + instance.Name + ".network.0.fixed_ip_v4}"
 	}
-	consulKeys.Keys = append(consulKeys.Keys, commons.ConsulKey{Path: path.Join(instancesKey, instanceName, "/capabilities/endpoint/attributes/ip_address"), Value: accessIP}) // Use access ip here
-
-	// Add Connection check
-	if err = commons.AddConnectionCheckResource(infrastructure, user, privateKey, accessIP, instance.Name, env); err != nil {
-		return err
-	}
 
-	consulKeyAttrib := commons.ConsulKey{Path: path.Join(instancesKey, instanceName, "/attributes/ip_address"), Value: fmt.Sprintf("${openstack_compute_instance_v2.%s.network.%d.fixed_ip_v4}", instance.Name, len(instance.Networks)-1)} // Use latest provisioned network for private access
-	consulKeyFixedIP := commons.ConsulKey{Path: path.Join(instancesKey, instanceName, "/attributes/private_address"), Value: fmt.Sprintf("${openstack_compute_instance_v2.%s.network.%d.fixed_ip_v4}", instance.Name, len(instance.Networks)-1)}
-	consulKeys.Keys = append(consulKeys.Keys, consulKeyAttrib, consulKeyFixedIP)
+	// Provide output for access IP and private IP
+	accessIPKey := nodeName + "-" + instanceName + "-publicIP"
```

Comment:
Looks like this line can redefine the public IP key and assign it the private IP value (when provisioning_over_fip_allowed is false). This is what I see in a deployment test on OpenStack as well where the public ip value is unexpectedly the private IP

Reply:
You're right. I fix it.

Reply:
Tested the new version, the public IP is correct